infrastructure/firecracker/provider.go

```diff
@@ -25,6 +26,8 @@ type Config struct {
 	StateRoot string
 	// RunDetached indicates that the firecracker processes should be run detached (a.k.a daemon) from the parent process.
 	RunDetached bool
+	// Timeout is the timeout for the firecracker process operations.
+	Timeout time.Duration
```

Comment:
Following on from @yitsushi comment, perhaps this should be a more specific `ShutdownTimeout` or something similar.